### ISSUE TITLE
Error in global aggregation for FedAsync and FedBuf for complex model

### ISSUE THREAD

**Issue by Zilinghan** [APPFL/APPFL#131]
When @ShellyRiver and I tested asynchronous federated learning algorithms using complex models such as ResNet or some baseline models in FLamby, all async FL algorithms had extremely low accuracy compared to synchronous ones. 
We finally realize that the problem occurs in the global aggregation process:
As shown in the FedSync code https://github.com/APPFL/APPFL/blob/9e58acde6212b6bbb311f0249cb6e00f3410e682/src/appfl/algorithm/server_federated.py#LL78C1-L86C27 
there may be model parameters in `self.model.state_dict()` but not in `self.model.named_parameters()`, but both asynchronous FL algorithms (FedAsync and FedBuf) did not account this, thus leading to low accuracy in complex models.
